IPython/utils/decorators.py

```diff
@@ -40,9 +40,8 @@ def flag_calls(func):
 
     def wrapper(*args,**kw):
         wrapper.called = False
-        out = func(*args,**kw)
         wrapper.called = True
-        return out
+        return func(*args,**kw)
```

Comment:
That one will not have the same behavior if `func(*args, **kwargs)` raises, and it's incorrect from the docstring:

```
determine if a call to
    func() was attempted and succeeded.
```